jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ThreadLimitHandler.java

```diff
@@ -305,10 +310,10 @@ protected Callback getCallback()
 
         protected void process() throws Exception
         {
-            CompletableFuture<Closeable> futurePermit = _remote.acquire();
+            Permit futurePermit = _remote.acquire();
 
             // Did we get a permit?
-            if (futurePermit.isDone())
+            if (futurePermit.isAllocated())
             {
                 if (LOG.isDebugEnabled())
                     LOG.debug("Threadpermitted {}", _remote);
```

Comment:
Missing a space between `Thread` and `permitted`. Maybe worth logging also the request and the current thread?